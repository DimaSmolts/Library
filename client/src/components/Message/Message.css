.message {
  margin: 15% auto;
  text-align: center;
}

.msgSmall {
  font-size: 3ex;
}
.msgMedium {
  font-size: 6ex;
}
.msgBig {
  font-size: 9ex;
}

.msgInfo,
.msgSuccess,
.msgNotFound {
  color: var(--dark-blue);
}

.msgError {
  color: red;
}
