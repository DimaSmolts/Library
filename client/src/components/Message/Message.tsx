import React from 'react';
import './Message.css'

export enum MessageTypes {
  msgError,
  msgInfo,
  msgSuccess,
  msgNotFound,
}

export enum MessageTextSizes {
  msgSmall,
  msgMedium,
  msgBig,
}

export interface MessageProps {
  text: string,
  type: MessageTypes
  size: MessageTextSizes
}

export default class Message extends React.Component<MessageProps> {
  render() {
    return (
      <div className='message-container'>
        <p className={
          'message ' +
          MessageTextSizes[this.props.size] + ' ' +
          MessageTypes[this.props.type]}>
          {this.props.text}
        </p>
      </div>
    )
  }
}
