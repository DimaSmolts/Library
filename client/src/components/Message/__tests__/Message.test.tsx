import React from 'react'
import renderer from 'react-test-renderer'

import Message, { MessageTypes, MessageTextSizes } from '../Message'
import { mount } from 'enzyme'

test('snapshot: <Message /> renders correctly', () => {
  const tree = renderer.create(
    <Message
      type={MessageTypes.msgInfo}
      size={MessageTextSizes.msgMedium}
      text='hey' />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

it('should map Message props to css classes correctly', () => {
  const message = mount(
    <Message
      type={MessageTypes.msgInfo}
      size={MessageTextSizes.msgMedium}
      text='hey' />)

  expect(message.find('div').hasClass('message-container')).toBeTruthy()
  expect(message.find('p').hasClass('message')).toBeTruthy()
  expect(message.find('p').hasClass('msgMedium')).toBeTruthy()
  expect(message.find('p').hasClass('msgInfo')).toBeTruthy()
  expect(message.find('p').text().includes('hey'))
})
