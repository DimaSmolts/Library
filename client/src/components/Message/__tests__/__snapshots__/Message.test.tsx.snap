// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: <Message /> renders correctly 1`] = `
<div
  className="message-container"
>
  <p
    className="message msgMedium msgInfo"
  >
    hey
  </p>
</div>
`;
